lighthouse-extension/app/src/lighthouse-background.js

```diff
@@ -124,19 +147,30 @@ window.runLighthouseForConnection = function(connection, url, options, requested
   lighthouseIsRunning = true;
   updateBadgeUI(url);
 
-  // Run Lighthouse.
-  return Runner.run(connection, runOptions)
-    .then(result => {
-      lighthouseIsRunning = false;
-      updateBadgeUI();
-      filterOutArtifacts(result);
-      return result;
-    })
-    .catch(err => {
-      lighthouseIsRunning = false;
-      updateBadgeUI();
-      throw err;
-    });
+  return enableOtherChromeExtensions(false)
+      .then(_ => Runner.run(connection, runOptions)) // Run Lighthouse.
+      .then(result => {
+        lighthouseIsRunning = false;
+        updateBadgeUI();
+        filterOutArtifacts(result);
+
+        // Deliver results even if enabling extensions fails.
+        enableOtherChromeExtensions(true).catch(err => {
+          log.warn('Chrome', `Could not disable some extensions. ${err.message}`);
+        });
+
+        return result;
+      })
+      .catch(err => {
+        lighthouseIsRunning = false;
+        updateBadgeUI();
+
+        enableOtherChromeExtensions(true).catch(err => {
+          log.warn('Chrome', `Could not enable/disable some extensions. ${err.message}`);
```

Comment:
It would be nice to know if/when the enabling/disabling can fail, but it may be risky to make that fatal. I really wish we had analytics :\

As part of your error refactoring, do you want to think about ways to surface top-level errors like this (e.g. non-gatherer errors) to the report so people can file bugs on them?